src/AppDevRouter.js

```diff
@@ -1,86 +1,113 @@
 // @flow
-// A REST-API Router parent that handles boilerplate for
-// serving up JSON responses based on HTTP verb
 
-import type {RequestType} from './constants';
+import AppDevUtilities from './AppDevUtilities';
+import { Router, Request, Response, NextFunction } from 'express';
 
-import {
-  Router,
-  Request,
-  Response,
-  NextFunction
-} from 'express';
+/**
+ * RequestType - the HTTP methods supported by AppDevRouter
+ */
+export type RequestType = 'GET' | 'POST' | 'DELETE';
 
-import constants from './constants';
-
-class AppDevResponse {
+/**
+ * AppDevResponse - the response from an HTTP request
+ * 
+ * Wraps a `success` field around the response data
+ */
+class AppDevResponse<T> {
   success: boolean;
-  data: Object;
+  data: T;
 
-  constructor (success: boolean, data: Object) {
+  constructor (success: boolean, data: T) {
     this.success = success;
     this.data = data;
   }
 }
 
-class AppDevRouter {
+/**
+ * AppDevRouter - cleanly create an Express Router object using inheritance
+ *
+ * Subclasses can simply specify the HTTP method, the path, and a response 
+ * hook to compute response data. This pattern is cleaner than raw Express 
+ * Router initialization with callbacks.
+ */
+class AppDevRouter<T> {
```

Comment:
Did not know you could parametrize –– very cool

Reply:
I learned this from clicker backend lmao